File: cpp/solved/7.cpp
#include <iostream>

using namespace std;


int prime(int n) {
    if (n == 2) {
        return 1;
    } else if (n < 2 || n%2 == 0) {
        return 0;
    } else {
        int i = 3;
        while (i*i <= n) {
            if (n%i == 0) {
                return 0;
            }
            i += 2;
        }
        return 1;
    }

}

int main() {
    int i = 2, c = 0, N = 10001;
    while (true) {
        if (prime(i) == 1) {
            c += 1;
            cout << i << "," << c << endl;
        }
        if (c == N) {
            break;
        }
        i += 1;
    }
    return 0;
}

File: py/solved/7.py
#!/bin/sh

# p7.py
# ProjectEuler
#
# Created by joe yuan on 3/8/11.
# Copyright 2011 __MyCompanyName__. All rights reserved.


def prime(f):
    prime = [2]
    #for i in range(3,int(f**.5)): #<----prime factors
    #for i in range(3,int(f**.5)) #<----modified for primes up to f
    y = 1003
    while len(prime) != f: #<----n number of primes
        for i in range(y-1000,y): #<---- indent everything below 
            #if i%1000 == 0:
            #    print(i, len(prime))
            for c in prime:
                if i%c == 0:
                    u = False
                    break
                u = True
            if u: 
                prime.append(i)
            if len(prime) == f:
                break
        y = y + 1000
    return prime

def solve():
    f = 10001
    p = prime(f)
    return p[-1]

from timer import time_function
print(time_function(solve))
